api_tests/actions/views/test_schema_response_action_list.py

```diff
@@ -0,0 +1,181 @@
+import pytest
+
+from osf_tests.factories import (
+    AuthUserFactory,
+    RegistrationFactory,
+    SchemaResponseFactory,
+    SchemaResponseActionFactory
+)
+
+from osf.utils.workflows import SchemaResponseTriggers, ApprovalStates
+
+@pytest.mark.django_db
+@pytest.mark.enable_quickfiles_creation
+class TestSchemaResponseActionList:
+
+    @pytest.fixture()
+    def user(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def registration(self):
+        return RegistrationFactory()
+
+    @pytest.fixture()
+    def schema_response(self, registration):
+        return SchemaResponseFactory(
+            registration=registration,
+            initiator=registration.creator,
+        )
+
+    @pytest.fixture()
+    def unapproved_schema_response(self, schema_response, user):
+        schema_response.approvals_state_machine.set_state(ApprovalStates.UNAPPROVED)
+        schema_response.pending_approvers.add(user)
+        schema_response.save()
+        return schema_response
+
+    @pytest.fixture()
+    def schema_response_action(self, schema_response):
+        return SchemaResponseActionFactory(
+            target=schema_response,
+            trigger=SchemaResponseTriggers.SUBMIT.db_name,
+            from_state=ApprovalStates.IN_PROGRESS.db_name,
+            to_state=ApprovalStates.APPROVED.db_name
+        )
+
+    def make_payload(self, schema_response, trigger):
+        return {
+            'data':
+                {
+                    'type': 'schema-response-actions',
+                    'attributes': {
+                        'trigger': trigger.db_name,
+                        'comment': "It's Party time!",
+                    },
+                    'relationships': {
+                        'target': {
+                            'data': {
+                                'id': schema_response._id,
+                                'type': 'schema-responses'
+                            }
+                        }
+                    }
+                }
+        }
+
+    @pytest.fixture()
+    def url(self, schema_response):
+        return f'/v2/schema_responses/{schema_response._id}/actions/'
+
+    def test_schema_response_action_submit(self, app, registration, schema_response, user, url):
+        assert schema_response.reviews_state == ApprovalStates.IN_PROGRESS.db_name
+        registration.add_contributor(user, 'admin')
+        assert not schema_response.pending_approvers.count()
+        payload = self.make_payload(schema_response=schema_response, trigger=SchemaResponseTriggers.SUBMIT)
+        resp = app.post_json_api(url, payload, auth=user.auth, expect_errors=True)
+        assert resp.status_code == 201
+        schema_response.refresh_from_db()
+        assert schema_response.reviews_state == ApprovalStates.UNAPPROVED.db_name
```

Comment:
check the list of pending_approvers here, please, since the serializer has the logic to produce that list